mcodingbot/utils/peps.py

```diff
@@ -39,10 +41,19 @@ async def fetch_pep_info(self, bot: Bot) -> None:
                     )
                     for pep_id, pep in (await resp.json()).items()
                 }
+                self._pep_map = {
+                    k: f"{v.title} ({v.number})" for k, v in self._peps.items()
+                }
 
     def get(self, pep_number: int) -> PEPInfo | None:
         return self._peps.get(pep_number)
 
+    def search(
+        self, query: str, limit: int = 20
+    ) -> Generator[PEPInfo, None, None]:
```

Comment:
This returns `typing.Iterable[PEPInfo]`

Reply:
No it returns a generator. Generators are also Iterables, but it's best to be specific with returns types and broad with input types.

Reply:
Keep it or change it?

Reply:
I say keep it, unless @Lunarmagpie has a good reason to change it

Reply:
Generator inherits from Iterable (check typeshed)
we dont need any of the added things. i do not believe they apply when its Generator[T, None, None]

Reply:
If generator inherits from Iterable then this should be a Generator, as that's what's being returned.

You wouldn't do this
```py
def function() -> object:
    return "hello"
```

You would use `str`, which is a subclass of `object`, as that is more specific. Your return types should be as specific as possible, and argument types should be as broad as possible.

Reply:
nobody returns a string to use it like an object...
i will correct myself `Iterator` is more accurate. `Generator[T, None, None`] does not give us a single thing that does shit over Iterator.  I'm sure throw, close, gi_code, gi_frame, gi_yieldfrom, and gi_running are real useful. Your argument assumes they are as useful as basic string methods.

Reply:
Generators behave completely differently from all other Iterables, as Generators are consumable.

Reply:
https://discord.com/channels/826875707418214451/851854179534372884/1019844894912675892

Reply:
@trag1c thoughts ~~save me~~

Reply:
> 

So use iterator. The docs recommend it as an alternative as well.

Reply:
> @trag1c thoughts? ~save me~

Use `Iterator[PEPInfo]`